pulsar-broker-common/src/main/java/org/apache/pulsar/bookie/rackawareness/BookieRackAffinityMapping.java

```diff
@@ -157,7 +158,9 @@ private void watchAvailableBookies() {
                 registrationClient.watchWritableBookies(versioned -> {
                     bookieMappingCache.get(BOOKIE_INFO_ROOT_PATH)
                             .thenApply(optRes -> optRes.orElseGet(BookiesRackConfiguration::new))
-                            .thenAccept(this::updateRacksWithHost)
+                            .thenApply(racks ->
+                                    processRackUpdate(racks, bookieAddressListLastTime)
```

Comment:
accessing `bookieAddressListLastTime` isn't thread safe. It seems that making the field `volatile` would address the issue.